modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+|===
+
+[id="gitops-components_{context}"]
+== Components
+
+Argo Rollouts consists of several components that enable users to practice Progressive Delivery on OpenShift.
+
+[options="header"]
+|===
+|Name | Description
+|`Rollouts Controller` | The Rollouts Controller component of the Argo Rollouts manages the Rollouts custom resource which resembles a native Kubernetes Deployment resource. The Rollouts Controller does not modify the standard deployment resources and only responds to changes in the Rollouts resources.
+|`AnalysisRun Controller` | The `AnalysisRun` controller performs analysis for an `AnalysisRun` resource and manages the `AnalysisRun` custom resource. It provides the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that decide if a deployment update is successful for your application.
+
+Argo Rollouts supports two custom Kubernetes resources to perform Analysis - `AnalysisTemplate` and `AnalysisRun`. You can define an `AnalysisTemplate` on a specific Rollout or globally on a cluster to be shared by multiple rollouts as a Cluster`AnalysisTemplate`. The `AnalysisRun` resource is scoped on a specific rollout.
+|`Experiment Controller` | The `Experiment` Controller creates analysis for limited-duration, short-lived replica sets, and manages the `Experiment` custom resource. The `Experiment` custom resource can be used independently or executed as part of Rollouts. You can link a list of `AnalysisTemplate` to be used on replica sets by creating an `Experiment` custom resource.
+
+In addition, this controller is used for creating replica sets. You can use it for optionally creating the service, running the analysis, reporting the results, and scaling down the replica sets in the end. The `Experiment` controller can also be integrated with Rollouts by specifying the experiment step in the canary deployment strategy.
+|`Service and Ingress Controller` | The `Service` controller handles the `Service` resources and the `Ingress` controller manages the `Ingress` resources modified by Argo Rollouts. These controllers inject additional metadata annotations in the application instances for traffic management.
+
+Argo Rollouts supports the standard Kubernetes Service and Ingress resources for forwarding the incoming user traffic to the correct version of an application. Depending on the deployment strategy and the traffic management tool used by your application, you can configure Argo Rollouts to split incoming traffic in application instances with specific percentages.
+|`UI and CLI` | Argo Rollouts supports an `oc/kubectl` plugin to interact with resources like Rollouts, Analysis, and Experiments from the command line. It can perform operations like pause, promote, retry, on the Rollout resource. The Argo Rollouts CLI plugin can start a local web UI dashboard in the browser to enhance the experience of visualizing the Rollouts resources.
+|===
```

Comment:
```suggestion
|===

```
Add one additional blank space to ensure the build will generate correctly.

Reply:
Fixed.